Event.php

```diff
@@ -186,10 +190,23 @@ public static function unregister($eventId, $hard = false)
      * @param   mixed   $callable    Callable.
      * @return  \Hoa\Event\Event
      */
-    public function attach($callable)
+    public function attach($callable, $priority = 0)
     {
-        $callable                              = xcallable($callable);
-        $this->_callable[$callable->getHash()] = $callable;
+        $callable                 = xcallable($callable);
+        $hash                     = $callable->getHash();
+        $this->_callable[$hash]   = $callable;
+        $this->_priorities[$hash] = (int) $priority;
+
+        uksort($this->_callable, function ($a, $b) {
+            $a = $this->_priorities[$a];
+            $b = $this->_priorities[$b];
+
+            /*if (70000 <= PHP_VERSION_ID) {
```

Comment:
Yup, first try to make compatibility with spaceship operator but it's parser issue.